src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherWithJobIntegrationTests.java

```diff
@@ -147,4 +153,91 @@ public void testJobAnnotations() {
 				Matchers.<TaskStatus>hasProperty("state", Matchers.is(LaunchState.unknown))), timeout.maxAttempts,
 				timeout.pause));
 	}
+
+
+	@Test
+	public void testJobSpecProperties() {
+		log.info("Testing {}...", "JobSpecProperties");
+
+		KubernetesDeployerProperties kubernetesDeployerProperties = new KubernetesDeployerProperties();
+		kubernetesDeployerProperties.setCreateJob(true);
+
+		KubernetesTaskLauncher kubernetesTaskLauncher = new KubernetesTaskLauncher(kubernetesDeployerProperties,
+				new KubernetesTaskLauncherProperties(), this.kubernetesClient);
+
+		AppDefinition definition = new AppDefinition(randomName(), null);
+		Resource resource = testApplication();
+		Map<String, String> appRequest = new HashMap<>();
+		appRequest.put("spring.cloud.deployer.kubernetes.restartPolicy", "OnFailure");
+		appRequest.put("spring.cloud.deployer.kubernetes.backoffLimit", "5");
+		AppDeploymentRequest request = new AppDeploymentRequest(definition, resource, appRequest);
+
+		log.info("Launching {}...", request.getDefinition().getName());
+
+		String launchId = kubernetesTaskLauncher.launch(request);
+		Timeout timeout = deploymentTimeout();
+
+		assertThat(launchId, eventually(hasStatusThat(
+				Matchers.<TaskStatus>hasProperty("state", Matchers.is(LaunchState.launching))), timeout.maxAttempts,
+				timeout.pause));
+
+		String taskName = request.getDefinition().getName();
+
+		log.info("Checking job spec of {}...", taskName);
+
+		List<io.fabric8.kubernetes.api.model.batch.Job> jobs = kubernetesClient.batch().jobs().withLabel("task-name", taskName).list().getItems();
+
+		assertThat(jobs.size(), is(1));
+
+		io.fabric8.kubernetes.api.model.batch.Job job = jobs.get(0);
+		assertTrue(job.getSpec().getBackoffLimit().equals(5));
```

Comment:
could just use `assertEquals` here (or assertThat to be consistent with the one above)